common/src/Microsoft.Azure.IIoT.Hub.Module.Client/src/Default/DeviceClientAdapter.cs

```diff
@@ -0,0 +1,238 @@
+            throw new NotSupportedException("Device client does not support methods");
+        }
+
+        /// <inheritdoc />
+        public void Dispose() {
+            IsClosed = true;
+            _client?.Dispose();
+        }
+
+        /// <summary>
+        /// Handle status change event
+        /// </summary>
+        /// <param name="deviceId"></param>
+        /// <param name="onConnectionLost"></param>
+        /// <param name="logger"></param>
+        /// <param name="status"></param>
+        /// <param name="reason"></param>
+        private void OnConnectionStatusChange(string deviceId,
+            Action onConnectionLost, ILogger logger, ConnectionStatus status,
+            ConnectionStatusChangeReason reason) {
+
+            if (status == ConnectionStatus.Connected) {
+                logger.Information("{counter}: Device {deviceId} reconnected " +
+                    "due to {reason}.", _reconnectCounter, deviceId, reason);
+                kReconnectionStatus.WithLabels(deviceId, DateTime.UtcNow.ToString()).Set(_reconnectCounter);
+                _reconnectCounter++;
+                return;
+            }
+            logger.Information("{counter}: Device {deviceId} disconnected " +
+                "due to {reason} - now {status}...", _reconnectCounter, deviceId,
+                    reason, status);
+            kDisconnectionStatus.WithLabels(deviceId, DateTime.UtcNow.ToString()).Set(_reconnectCounter);
+            if (IsClosed) {
+                // Already closed - nothing to do
+                return;
+            }
+            if (status == ConnectionStatus.Disconnected ||
+                status == ConnectionStatus.Disabled) {
+                // Force
+                IsClosed = true;
+                onConnectionLost?.Invoke();
+            }
+        }
+
+        /// <summary>
+        /// Helper to create device client
+        /// </summary>
+        /// <param name="cs"></param>
+        /// <param name="transportSetting"></param>
+        /// <returns></returns>
+        private static DeviceClient Create(IotHubConnectionStringBuilder cs,
+            ITransportSettings transportSetting) {
+            if (cs == null) {
+                throw new ArgumentNullException(nameof(cs));
+            }
+            if (transportSetting != null) {
+                return DeviceClient.CreateFromConnectionString(cs.ToString(),
+                    new ITransportSettings[] { transportSetting });
+            }
+            return DeviceClient.CreateFromConnectionString(cs.ToString());
+        }
+
+        private readonly DeviceClient _client;
+        private readonly ILogger _logger;
+
+        private int _reconnectCounter;
+        private static readonly Gauge kReconnectionStatus = Metrics
+            .CreateGauge("iiot_edge_device_reconnected", "reconnected count",
+                new GaugeConfiguration {
+                    LabelNames = new[] { "device", "timestamp_utc" }
```

Comment:
having a timestamp as a label is a bad practice, please remove.

Reply:
Done